container/Dockerfile

```diff
@@ -53,3 +53,32 @@ USER karapace
 
 HEALTHCHECK --interval=10s --timeout=30s --retries=3 --start-period=60s \
     CMD python3 healthcheck.py http://localhost:$KARAPACE_PORT/_health || exit 1
+
+
+FROM builder AS dev-builder
+
+# Install dev dependencies.
+COPY ./requirements/requirements-dev.txt /build/
+COPY ./requirements/requirements-typing.txt /build/
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python3 -m pip install -r /build/requirements-dev.txt -r /build/requirements-typing.txt
+
+
+# Dev build stage
+FROM karapace AS dev
+
+# Install linux dependencies
+USER root
+
+RUN apt-get update \
+ && apt-get install --assume-yes --no-install-recommends \
+    openjdk-11-jdk \
+ && rm -rf /var/lib/apt/lists/*
+
+USER karapace
+
+
+COPY --from=dev-builder /venv /venv
+
+RUN mkdir /opt/karapace/app
+ENV PYTHONPATH="/opt/karapace/app:${PYTHONPATH}"
```

Comment:
issue: I'd prefer if we maintain the same way of making source code available to the interpreter under both targets. Can we make the dev target mount the app under the same path as it is mounted in the prod (`karapace`) target, and use the same working directory?

Reply:
> Can we make the dev target mount the app under the same path as it is mounted in the prod (karapace) target, and use the same working directory?

Not very easy afaik, the issue is that on the `karapace` target we copy the utility scripts e.g. `healthcheck.py` and `start.sh` into that folder. We don't make the source code available in the `karapace` target, the code is installed with `setup.py` and then only goes to the `site-packages` folder under `/usr/lib`